torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -772,10 +773,15 @@ std::shared_ptr<ProcessGroupNCCL::WorkNCCL> ProcessGroupNCCL::initWork(
 
 c10::intrusive_ptr<c10::ivalue::Future> ProcessGroupNCCL::WorkNCCL::
     getFuture() {
-  // FutureNCCL has a reference to WorkNCCL. Therefore, we don't store a
-  // FutureNCCL reference inside WorkNCCL and we create a new object here
-  // to avoid circular reference between WorkNCCL and FutureNCCL.
-  return c10::make_intrusive<FutureNCCL>(shared_from_this(), outputs_);
+  TORCH_INTERNAL_ASSERT(
+      outputs_->size() == 1,
+      "WorkNCCL's getFuture API is only supported for single-process single-device mode.");
+  // Create a new FutureNCCL object after checking for single-process
+  // single-device mode.
+  std::shared_ptr<at::cuda::CUDAEvent> cudaEvent = {cudaEvents_,
+                                                    &(*cudaEvents_)[0]};
```

Comment:
Why are we doing this and is this even correct? Shouldn't we just pass `cudaEvents_` to FutureNCCL?

Reply:
I was trying to address, @rohan-varma's comment https://github.com/pytorch/pytorch/pull/42869#discussion_r468863368. 

We just have one event, so probably using a vector isn't needed. I was just trying to create a pointer that shares the ownership with `cudaEvents_` and points to its first element. Since `at::cuda::CUDAEvent`'s constructor is deleted, I need to be careful here, but probably that's not an elegant solution and linter gives warning.

If using a vector is okay, I can just revert this and keep using vector inside `FutureNCCL`.

Reply:
I think let's just pass a vector since I'm not sure if it's possible to create a shared_ptr pointing to one element of a shared_ptr vector.